plugins/examples/dynamic-clusters/src/index.tsx

```diff
@@ -1,24 +1,102 @@
-import { Headlamp, registerAppBarAction } from '@kinvolk/headlamp-plugin/lib';
-import { Button } from '@mui/material';
+import { ApiProxy, Headlamp, registerAppBarAction } from '@kinvolk/headlamp-plugin/lib';
+import { Button, Modal, Paper, Stack, TextField, Typography } from '@mui/material';
+import { useEffect, useState } from 'react';
+
+const request = ApiProxy.request;
+const modalStyle = {
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+  height: '100vh',
+};
 
 function ClusterCreationButton() {
-  return (
-    <Button
-      onClick={() => {
-        Headlamp.setCluster({
-          name: 'my-plugin-set-cluster',
-          server: 'http://phony.abcde-example-k8s-cluster.k8scluster',
+  const [open, setOpen] = useState(false);
+  const [kubeconfig, setKubeconfig] = useState('');
+  const [name, setName] = useState('');
+  const [server, setServer] = useState('');
+  const [isDynamicClusterEnabled, setIsDynamicClusterEnabled] = useState(false);
+
+  useEffect(() => {
+    // Fetch configuration from the backend when the component mounts
+    request('/config', {}, false, false)
+      .then((response: any) => {
+        setIsDynamicClusterEnabled(response.isDynamicClusterEnabled);
+      })
+      .catch((error: any) => {
+        console.error('Error fetching config:', error);
+      });
+  }, []);
+
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+
+  const handleSubmit = () => {
+    if (isDynamicClusterEnabled) {
+      // Handle kubeconfig submission
+      Headlamp.setCluster({ kubeconfig })
+        .then(() => {
+          window.location.reload();
         })
-          .then(() => {
-            window.location.reload();
-          })
-          .catch(e => {
-            console.log('Error setting cluster', e);
-          });
-      }}
-    >
-      New cluster
-    </Button>
+        .catch((e: any) => {
+          console.error('Error setting cluster', e);
```

Comment:
Could you please display this error to the user?

That will let me test the error handling more easily.

Reply:
Updated, but please remember, when stateless cluster is enabled, there won't be any response from the backend, because everything is done on the frontend. I did however have added checks to validate kubeconfig.